src/main/java/io/snyk/jenkins/tools/SnykInstaller.java

```diff
@@ -82,30 +84,27 @@ private boolean isUpToDate(FilePath expectedLocation) throws IOException, Interr
     return timestampDifference < updateInterval;
   }
 
-  private FilePath downloadSnykBinaries(FilePath expected, Node node, TaskListener log) throws IOException, InterruptedException {
-    LOG.info("Install Snyk version '{}' as single binary on node '{}'", version, node.getDisplayName());
-
-    final VirtualChannel nodeChannel = node.getChannel();
-    if (nodeChannel == null) {
-      throw new IOException(format("Node '%s' is offline", node.getDisplayName()));
-    }
+  private FilePath downloadSnykBinaries(FilePath expected, Node node) {
+    try {
+      LOG.info("Installing Snyk '{}' on Build Node '{}'", version, node.getDisplayName());
 
-    Platform platform = nodeChannel.call(new GetPlatform(node.getDisplayName()));
+      final VirtualChannel nodeChannel = node.getChannel();
+      if (nodeChannel == null) {
+        throw new IOException(format("Build Node '%s' is offline.", node.getDisplayName()));
+      }
 
-    try {
+      Platform platform = nodeChannel.call(new GetPlatform());
       URL snykDownloadUrl = DownloadService.getDownloadUrlForSnyk(version, platform);
       URL snykToHtmlDownloadUrl = DownloadService.getDownloadUrlForSnykToHtml("latest", platform);
       expected.mkdirs();
       nodeChannel.call(new Downloader(snykDownloadUrl, expected.child(platform.snykWrapperFileName)));
       nodeChannel.call(new Downloader(snykToHtmlDownloadUrl, expected.child(platform.snykToHtmlWrapperFileName)));
       expected.child(INSTALLED_FROM).write(snykDownloadUrl.toString(), UTF_8.name());
       expected.child(TIMESTAMP_FILE).write(valueOf(Instant.now().toEpochMilli()), UTF_8.name());
-    } catch (Exception ex) {
-      log.getLogger().println("Snyk Security tool could not installed: " + ex.getMessage());
-      throw new ToolDetectionException("Could not install Snyk CLI from binary", ex);
+      return expected;
+    } catch (RuntimeException | IOException | InterruptedException ex) {
+      throw new RuntimeException("Failed to install Snyk.", ex);
```

Comment:
Moved the try/catch to cover the entire method so we can have some shared context on any failures.